src/crypto/password/argon2.clj

```diff
@@ -0,0 +1,37 @@
+(ns crypto.password.argon2
+ "Functions for encrypting passwords using the recommended argon2 algorithm.
+
+  See: https://infosecscout.com/best-algorithm-password-storage
+  https://github.com/phxql/argon2-jvm"
+ (:import [de.mkammerer.argon2 Argon2 Argon2Factory Argon2Advanced]))
+
+(def argon2 (Argon2Factory/create))
+
+(def ^:private default-iterations
+  (Long/parseLong (System/getProperty "crypto.password.argon2.default-iterations" "22")))
+
+(def ^:private default-memory-cost
+  (Long/parseLong (System/getProperty "crypto.password.argon2.default-memory-cost" "65536")))
+
+(def ^:private default-parallelization-parameter
+  (Long/parseLong (System/getProperty "crypto.password.argon2.default-parallelization-parameter" "1")))
+
+
+(defn encrypt
+  "Encrypt a password string using the argon2 algorithm. This function takes
+  three optional parameters:
+
+  * `iter` - the number of iterations, defaults to 22 (see:
+  Argon2Helper/findIterations)
+  * `mem` - the memory cost, defaults to 65536
+  * `parallel` - the parallelization parameter, defaults to 1"
```

Comment:
I don't think the default needs to be justified in the docstring. Better to just keep the relevant information for usage, and line up the definitions for clarity. So:

```
  * `iter`     - the number of iterations, defaults to 22
  * `mem`      - the memory cost, defaults to 65536
  * `parallel` - the parallelization parameter, defaults to 1
```